rs/nervous_system/integration_tests/tests/sns_topics.rs

```diff
@@ -0,0 +1,196 @@
+        .with_nns_subnet()
+        .with_sns_subnet()
+        .build_async()
+        .await;
+
+    // Install the NNS canisters.
+    {
+        let mut nns_installer = NnsInstaller::default();
+        nns_installer.with_current_nns_canister_versions();
+        nns_installer.install(&pocket_ic).await;
+    }
+
+    // Publish SNS Wasms to SNS-W.
+    let with_mainnet_sns_canisters = false;
+    add_wasms_to_sns_wasm(&pocket_ic, with_mainnet_sns_canisters)
+        .await
+        .unwrap();
+
+    let sns = {
+        let create_service_nervous_system = CreateServiceNervousSystemBuilder::default().build();
+
+        let sns_instance_label = "1";
+        let (sns, _) = nns::governance::propose_to_deploy_sns_and_wait(
+            &pocket_ic,
+            create_service_nervous_system,
+            sns_instance_label,
+        )
+        .await;
+
+        sns::swap::await_swap_lifecycle(&pocket_ic, sns.swap.canister_id, Lifecycle::Open)
+            .await
+            .unwrap();
+
+        sns
+    };
+
+    // Get an ID of an SNS neuron that can submit proposals. We rely on the fact that this
+    // neuron either holds the majority of the voting power or the follow graph is set up
+    // s.t. when this neuron submits a proposal, that proposal gets through without the need
+    // for any voting.
+    let (sns_neuron_id, sender) = sns::governance::find_neuron_with_majority_voting_power(
+        &pocket_ic,
+        sns.governance.canister_id,
+    )
+    .await
+    .expect("cannot find SNS neuron with dissolve delay over 6 months.");
+
+    let pocket_ic_agent = PocketIcAgent {
+        pocket_ic: &pocket_ic,
+        sender: sender.into(),
+    };
+
+    let governance_canister = GovernanceCanister {
+        canister_id: sns.governance.canister_id,
+    };
+
+    let initial_generic_function = GenericNervousSystemFunction {
+        target_canister_id: Some(PrincipalId::new_user_test_id(42)),
+        target_method_name: Some("do_things".to_string()),
+        validator_canister_id: Some(PrincipalId::new_user_test_id(43)),
+        validator_method_name: Some("validate_things".to_string()),
+        topic: Some(Topic::DaoCommunitySettings),
+    };
+
+    let expected_generic_function = GenericNervousSystemFunction {
+        topic: Some(Topic::ApplicationBusinessLogic),
+        ..initial_generic_function.clone()
+    };
+
+    let initial_function = NervousSystemFunction {
+        id: 1111,
+        name: "Test Custom Proposal Type".to_string(),
+        description: Some("This is a custom proposal type for testing.".to_string()),
+        function_type: Some(FunctionType::GenericNervousSystemFunction(
+            initial_generic_function,
+        )),
+    };
+
+    let expected_function = NervousSystemFunction {
+        function_type: Some(FunctionType::GenericNervousSystemFunction(
+            expected_generic_function,
+        )),
+        ..initial_function.clone()
+    };
+
+    // AddGenericNervousSystemFunction
```

Comment:
I believe this comment can and should contain prose, since that is the standard way.